### PR TITLE
test(mcp): add unit tests for router/responses.ts

### PR DESCRIPTION
**Pull request by vijaygopalbalasa**
## Summary

Adds 112 unit tests for `src/router/responses.ts`, which previously had zero test coverage. This module is critical to the summary-first response pipeline and contains markdown-dependent patterns that could silently under-match if handler output changes (as noted in the source comments at lines 5-17).

## What's covered

| Function | Tests | Category |
|----------|-------|----------|
| `cleanText` | 7 | Text normalization (CRLF, newline collapsing, trimming) |
| `truncateText` | 10 | Boundary-aware truncation (heading, separator, threshold logic) |
| `compactErrorText` | 7 | Error text cleaning (JSON stripping, case-insensitivity, default limit) |
| `getFamilyLimit` | 5 | Limit lookup across all 8 families and 3 detail levels |
| `toPlainText` | 8 | Response content extraction (text filtering, structuredContent fallback) |
| `buildTextVariant` | 14 | Full variant pipeline (all families, all detail levels, all strip markers) |
| `estimateRenderedSize` | 5 | Size estimation edge cases |
| `collectSectionHints` | 6 | Heading extraction (h1-h3 capture, h4+ exclusion, limit enforcement) |
| `applySectionSelection` | 5 | Section filtering (positive match, no-match fallback, case-insensitive) |
| `applyItemSelection` | 7 | Item extraction (separator, bold, emoji splits, boundary errors) |
| `applyRangeSelection` | 10 | Range slicing (valid ranges, invalid format, edge cases) |
| `mcpText` | 5 | Response builder shape and meta handling |
| `mcpErrorCompact` | 3 | Error response builder |
| `mcpResultHandle` | 6 | Result handle with expansion options |

**Before:** 111 tests across 5 files
**After:** 223 tests across 6 files

## Test highlights

- All 8 `ResponseFamily` types tested at `summary`, `standard`, and `full` detail levels
- Boundary-aware truncation tested around the 55% threshold (verifies cut position, not just presence of `[truncated]`)
- All 6 `stripExampleCode` markers exercised independently (`Example Code`, `Example`, `typescript`, `ts`, `javascript`, `js`)
- Emoji-prefixed and bold-prefixed list item splitting validated
- Positive-path section extraction verified (not just the fallback)
- Case-insensitive JSON block stripping tested
- Edge cases: empty input, whitespace-only, out-of-range indices, invalid range format, floating-point ranges

## How to verify

```bash
cd helius-mcp
pnpm test
```

### CODE REVIEW

File: helius-mcp/tests/responses.test.ts [hunk 1]
```diff
@@ -0,0 +1,658 @@
...
+    'history',
+    'document',
+    'streamingConfig',
+    'catalog',
+  ])('handles %s family at summary detail', (family) => {
+    const text = longText(1000);
+    const limit = getFamilyLimit(family, 'summary');
+    const result = buildTextVariant(family, 'summary', text);
+    // Result should be within limit + [truncated] suffix
+    expect(result.length).toBeLessThanOrEqual(limit + 15);
```
Comment (0xIchigo) [inline helius-mcp/tests/responses.test.ts]: `truncateText` appends `\n\n[truncated]` but `buildTextVariant` tests use `+ 15` as tolerance. The actual suffix is 13 chars, so this should update this to prevent masking a 1-2 char regression

File: helius-mcp/tests/responses.test.ts [hunk 2]
```diff
@@ -0,0 +1,658 @@
...
+  });
+
+  it('collapses 3+ consecutive newlines', () => {
+    const result = compactErrorText('line1\n\n\n\nline2');
+    expect(result).not.toContain('\n\n\n');
+    expect(result).toContain('line1');
+    expect(result).toContain('line2');
+  });
+
+  it('uses default maxChars of 1200', () => {
```
Comment (0xIchigo) [inline helius-mcp/tests/responses.test.ts:138]: Since `truncateText` can snap to a boundary before the limit, the result might be significantly shorter. The test passes, but doesn't actually verify the default value is 1200—it only verifies that truncation occurs. We should consider asserting `result.length > 1100` here as a lower bound too

File: helius-mcp/tests/responses.test.ts [hunk 3]
```diff
@@ -0,0 +1,658 @@
...
+describe('applySectionSelection', () => {
+  it('returns full text when section is undefined', () => {
+    expect(applySectionSelection('full document', undefined)).toBe('full document');
+  });
+
+  it('returns full text when section is empty string', () => {
+    expect(applySectionSelection('full document', '')).toBe('full document');
+  });
+
+  it('extracts matching section content (positive path)', () => {
```
Comment (0xIchigo) [inline helius-mcp/tests/responses.test.ts:471]: We should also verify that the result does not contain `## Support` (i.e., the heading itself), since `extractSections` should stop at the next same-level heading

File: helius-mcp/tests/responses.test.ts [hunk 4]
```diff
@@ -0,0 +1,658 @@
...
+  it('treats items with undefined type as text', () => {
+    expect(toPlainText({ content: [{ text: 'implicit' }] })).toBe('implicit');
+  });
+
+  it('uses empty string for items with undefined text', () => {
+    const response = { content: [{ type: 'text' }] };
+    expect(toPlainText(response)).toBe('');
+  });
+
+  it('falls back to stringified structuredContent', () => {
```
Comment (0xIchigo) [inline helius-mcp/tests/responses.test.ts:257]: We should also be verifying the pretty-print format here (i.e., `null, 2`)

**Review by 0xIchigo — CHANGES_REQUESTED** [review]
Thank you for your contribution! There are just a few nits before we can merge this one

**Comment by vijaygopalbalasa** [comment]
Addressed all four items. Updated the suffix comment, added a lower bound assertion, verified the pretty print format with null and 2 indentation, and added an explicit heading exclusion check. Thanks for the detailed review.

**Review by 0xIchigo — APPROVED** [review]
Thank you for your contribution! 🚀
